add-on/panels/js/panel.jsx

```diff
@@ -891,6 +883,103 @@ loop.panel = (function(_, mozL10n) {
     }
   });
 
+  var SharePanelView = React.createClass({
+    mixins: [
+      Backbone.Events,
+      sharedMixins.DocumentVisibilityMixin,
+      sharedMixins.WindowCloseMixin
+    ],
+
+    propTypes: {
+      dispatcher: React.PropTypes.instanceOf(loop.Dispatcher).isRequired,
+      // Force render for testing purpose
+      forceRender: React.PropTypes.bool,
+      store: React.PropTypes.instanceOf(loop.store.RoomStore).isRequired
+    },
+
+    getInitialState: function() {
+      return this.props.store.getStoreState();
+    },
+
+    componentWillMount: function() {
+      loop.request("GetLoopPref", "facebook.enabled").then(result => {
+        this.setState({
+          facebookEnabled: result
+        });
+      });
+    },
+
+    componentDidMount: function() {
+      this.listenTo(this.props.store, "change", this._onStoreStateChanged);
+    },
+
+    componentWillUnmount: function() {
+      this.stopListening(this.props.store);
+    },
+
+    _onStoreStateChanged: function() {
+      this.setState(this.props.store.getStoreState());
+    },
+
+    onDocumentHidden: function() {
+      this.state.shouldOpenSharePanel && this._closeSharePanel();
+    },
+
+    componentWillUpdate: function(nextProps, nextState) {
+      // If we've just created a room, close the panel - the store will open
+      // the room.
+      if (this.state.pendingCreation &&
+          !nextState.pendingCreation && !nextState.error) {
+        this.setState({
+          shouldOpenSharePanel: true
+        });
+      }
+    },
+
+    _closeSharePanel: function() {
```

Comment:
Actually, I take that back, `_closeSharePanel` is reasonable appropraite. I think I'd go more for `_closePanel` or `handleClosePanel` and also move the `this.closeWindow()` into this function - it wasn't obvious that this function was going to do that as well.